controllers/nodefeaturediscovery_controller.go

```diff
@@ -37,29 +37,60 @@ import (
 	nfdv1 "github.com/kubernetes-sigs/node-feature-discovery-operator/api/v1"
 )
 
+// log is used to set the logger with a name that describes the actions of
+// functions and arguments in this file
 var log = logf.Log.WithName("controller_nodefeaturediscovery")
 
+// nfd is an NFD object that will be used to initialize the NFD operator
 var nfd NFD
 
 // NodeFeatureDiscoveryReconciler reconciles a NodeFeatureDiscovery object
 type NodeFeatureDiscoveryReconciler struct {
+
+	// client.Client reads and writes directly from/to the OCP API server. This
+	// field needs to be added to the reconciler because it is responsible for
+	// fetching objects from the server, which the NFD operator needs to do in
+	// order to add its labels to each node in the cluster.
 	client.Client
-	Log       logr.Logger
-	Scheme    *runtime.Scheme
-	Recorder  record.EventRecorder
+
+	// Log is used to log the reconciliation. Every controllers needs this.
+	Log logr.Logger
+
+	// Scheme is used by the kubebuilder library to set OwnerReferences. Every
+	// controller needs this.
+	Scheme *runtime.Scheme
+
+	// Recorder defines interfaces for working with OCP event recorders. This
+	// field is needed by NFD in order for NFD to write events.
+	Recorder record.EventRecorder
+
+	// AssetsDir defines the directory with assets under the operator image
 	AssetsDir string
 }
 
-// SetupWithManager sets up the controller with the Manager.
+// SetupWithManager sets up the controller with the Manager to create the controller.
+// The Manager serves the purpose of initializing shared dependencies (like caches
+// and clients) from the 'client.Client' field in the NodeFeatureDiscoveryReconciler
+// struct.
```

Comment:
Again, I'd leave out the names of vars/types

```suggestion
// SetupWithManager sets up the controller with a specified manager responsible for
// initializing shared dependencies (like caches and clients)
```